iaso/api/stocks/serializers.py

```diff
@@ -0,0 +1,397 @@
+        fields = ["id", "name"]
+        ref_name = "ProjectNestedSerializerForStockKeepingUnit"
+
+
+class OrgUnitTypeNestedSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = OrgUnitType
+        fields = ["id", "name"]
+        ref_name = "OrgUnitTypeNestedSerializerForStockKeepingUnit"
+
+
+class OrgUnitNestedSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = OrgUnit
+        fields = ["id", "name"]
+        ref_name = "OrgUnitNestedSerializerForStockKeepingUnit"
+
+
+class UserNestedSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ["id", "username"]
+        ref_name = "UserNestedSerializerForStockKeepingUnit"
+
+
+class FormNestedSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Form
+        fields = ["id", "name"]
+        ref_name = "FormNestedSerializerForStockKeepingUnit"
+
+
+class StockKeepingUnitSerializer(serializers.ModelSerializer):
+    projects = ProjectNestedSerializer(many=True)
+    org_unit_types = OrgUnitTypeNestedSerializer(many=True)
+    forms = FormNestedSerializer(many=True)
+    created_at = TimestampField()
+    created_by = UserNestedSerializer()
+    updated_at = TimestampField()
+    updated_by = UserNestedSerializer()
+    children = StockKeepingUnitChildrenSerializer(source="sku_children_parent", many=True)
+    deleted_at = TimestampField()
+
+    class Meta:
+        model = StockKeepingUnit
+        fields = [
+            "id",
+            "name",
+            "short_name",
+            "projects",
+            "org_unit_types",
+            "forms",
+            "children",
+            "display_units",
+            "display_precision",
+            "created_at",
+            "created_by",
+            "updated_at",
+            "updated_by",
+            "deleted_at",
+        ]
+        read_only_fields = ["created_at", "updated_at", "created_by", "updated_by", "deleted_at"]
+
+
+class StockKeepingUnitWriteSerializer(serializers.ModelSerializer):
+    projects = PrimaryKeyRelatedField(
+        many=True,
+        queryset=Project.objects.all(),
+        required=False,
+    )
+    org_unit_types = PrimaryKeyRelatedField(
+        many=True,
+        queryset=OrgUnitType.objects.all(),
+        required=False,
+    )
+    forms = PrimaryKeyRelatedField(
+        many=True,
+        queryset=Form.objects.all(),
+        required=False,
+    )
+
+    class Meta:
+        model = StockKeepingUnit
+        fields = [
+            "id",
+            "name",
+            "short_name",
+            "projects",
+            "org_unit_types",
+            "forms",
+            "display_units",
+            "display_precision",
+        ]
+
+    def validate(self, validated_data):
+        request = self.context.get("request")
+        if "projects" in validated_data and not all(
+            p.account == request.user.iaso_profile.account for p in validated_data["projects"]
```

Comment:
This will do the exact same thing, but might avoid fetching the relation 
```suggestion
            p.account_id == request.user.iaso_profile.account_id for p in validated_data["projects"]
```